File: Brute Force/leetcode_01.cpp

// Problem Link ====>>Problem Link ====>> https://leetcode.com/problems/two-sum/description/

#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        int n = nums.size();
        for (int i = 0; i < n - 1; i++)
        {
            for (int j = i + 1; j < n; j++)
            {
                if (nums[i] + nums[j] == target)
                {
                    return {i, j};
                }
            }
        }
        return {}; // No solution found
    }
};

int main()
{
    Solution solution;
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;

    vector<int> result = solution.twoSum(nums, target);

    if (!result.empty())
    {
        cout << "Indices of the two numbers that add up to the target: " << result[0] << " " << result[1] << endl;
    }
    else
    {
        cout << "No solution found." << endl;
    }

    return 0;
}

File: Brute Force/leetcode_01.py

# Problem Link ====>> https://leetcode.com/problems/two-sum/description/

# Solution 1: (Brute Force)
def two_sum(nums, target):
    seen = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in seen:
            return [seen[complement], i]
        seen[num] = i
    return []


nums = [2, 7, 11, 15]
target = 9

result = two_sum(nums, target)

if result:
    print("Indices of the two numbers that add up to the target:",
          result[0], result[1])
else:
    print("No solution found.")


